src/main/java/org/broadinstitute/consent/http/authentication/AuthorizationHelper.java

```diff
@@ -49,6 +49,11 @@ protected AuthUser buildAuthUserFromHeaders(Map<String, String> headers) {
       logWarn(String.format(
           "Reading oauth2 claim headers: email is null, auth user is incomplete. Aud: %s Name: %s",
           aud, name));
+    } else {
+      User user = userService.findUserByEmail(email);
```

Comment:
This will throw an NFE which will be problematic for authenticated identities that do not yet exist in the system

Reply:
More thoughts on this code path. Another usage of this method is in `DuosUserAuthenticator` where we also make the same call to `userService.findUserByEmail(email)` so we'd be making the call twice for `@Auth DuosUser` usages. When resources use `@Auth AuthUser`, we usually follow up with a call to `userService.findUserByEmail()'. I wonder if we called the enforcement there would be more efficient.

Reply:
Yes, that is true - forgot about that one 👍🏽